src/core/inductive.ml

```diff
@@ -76,47 +132,56 @@ let gen_ind_typ_codom : popt -> sym -> (tbox list -> tbox) -> string -> tbox =
          [codom ts xs acc].
       2) If the current type is a product of the form [Π(x:ind_sym ts), b],
          then we are in case of recursive occurrences, so the function
-         [build_rec_hyp ts x] builds the recursive hypothesis associated, and
+         [build_rec_hyp s p ts x] builds the recursive hypothesis associated,
          then the function [domrec a x rec_hyp next] is applied to conclude
          the building ([rec_hyp] is the current recursive hypothesis and
          [next] is the result of the recursive call).
       3) If the current type is a product [Π(x:a), b] not of the previous
          form, then the function [dom a x next] is called.
-      4) Any other case is an error.*)
-let fold_cons_typ (pos : popt) (codom : term list -> 'b list -> 'a -> 'c)
+      4) Any other case is an error. *)
+let fold_cons_typ (pos : popt)
+      (codom : tvar -> term list -> 'b list -> 'a -> 'c)
       (inj_var : 'b list -> tvar -> 'b)
-      (build_rec_hyp : term list -> 'b -> 'a)
+      (build_rec_hyp : sym -> tvar -> term list -> 'b -> 'a)
       (domrec : term -> 'b -> 'a -> 'c -> 'c) (dom : term -> 'b -> 'c -> 'c)
       (ind_sym : sym) (cons_sym : sym) (f_rec : 'a -> 'b -> 'a -> 'a)
-      (f_not_rec : 'a -> 'b -> 'a) (init : 'a) (s : string) : 'c =
+      (f_not_rec : 'a -> 'b -> 'a) (init : 'a) (s : string)
+      (assoc_predicates : (sym * (tvar * tbox)) list) : 'c =
```

Comment:
It seems that only the tvar part is used in this function, that is, we could have `assoc_predicates : (sym * tvar) list`. To allow more arguments, we could have `assoc_predicates : (sym * (tvar * `something_else)) list`.

Reply:
This does not correspond to the rules that I had set for myself: fixed type when it is the same for the 2 instances, abstract with 'a,' b otherwise.

Reply:
In this function, do you use the tbox part or not?

Reply:
It's not important but OK.

Reply:
Yes it is because then we change replace tbox by tbox*tbox or anything else.

Reply:
Please do not mark conversations as resolved if they are not.

Reply:
It's solved : I've done the change.